crates/burn-jit/src/kernel/conv/conv2d/tune/conv2d.rs

```diff
@@ -173,19 +145,39 @@ fn should_run<R: JitRuntime, F: FloatElement>(
         // GEMM balanced
         4 => check_algo!(Balanced, F, &op.input, &conv_problem),
         _ => true,
-    }
+    };
+
+    println!("{index} => {should_run}");
+    should_run
 }
 
 fn can_launch<S: ConvSelector<ImplicitCmmaConv>, R: JitRuntime, CS: ConvPrecision>(
     input: &JitTensor<R>,
     conv_problem: &ConvolutionProblem,
 ) -> bool {
-    let plane_dim = 32;
+    let plane_dim = match input
+        .client
+        .properties()
+        .hardware_properties()
+        .defined_plane_size()
+    {
+        Some(val) => val,
+        None => return false,
+    };
 
     let (selection, config_input) = S::select_kernel::<R, CS>(plane_dim);
     let cube_dim = ImplicitCmmaConv::cube_dim(&selection);
     let cube_count = ImplicitCmmaConv::cube_count(&selection, conv_problem);
 
+    let max_cube_dim = 65535;
+
+    if let cubecl::CubeCount::Static(x, y, z) = cube_count {
+        println!("({x}, {y}, {z})");
+        if x > max_cube_dim || y > max_cube_dim || z > max_cube_dim {
+            return false;
+        }
+    }
```

Comment:
```suggestion
    let max_cube_dim = 65535; // This is the maximum cube count on wgpu

    if let cubecl::CubeCount::Static(x, y, z) = cube_count {
        if x > max_cube_dim || y > max_cube_dim || z > max_cube_dim {
            return false;
        }
    }
```

Reply:
Also, it would be better if this was either a constant or query from the device.